include/libopencm3/nrf5/common/gpio_common.h

```diff
@@ -0,0 +1,966 @@
+#define GPIO_LATCH_PIN19_Latched                  (1 << 19)
+#define GPIO_LATCH_PIN20_NOTLATCHED               (0 << 20)
+#define GPIO_LATCH_PIN20_NotLatched               (0 << 20)
+#define GPIO_LATCH_PIN20_LATCHED                  (1 << 20)
+#define GPIO_LATCH_PIN20_Latched                  (1 << 20)
+#define GPIO_LATCH_PIN21_NOTLATCHED               (0 << 21)
+#define GPIO_LATCH_PIN21_NotLatched               (0 << 21)
+#define GPIO_LATCH_PIN21_LATCHED                  (1 << 21)
+#define GPIO_LATCH_PIN21_Latched                  (1 << 21)
+#define GPIO_LATCH_PIN22_NOTLATCHED               (0 << 22)
+#define GPIO_LATCH_PIN22_NotLatched               (0 << 22)
+#define GPIO_LATCH_PIN22_LATCHED                  (1 << 22)
+#define GPIO_LATCH_PIN22_Latched                  (1 << 22)
+#define GPIO_LATCH_PIN23_NOTLATCHED               (0 << 23)
+#define GPIO_LATCH_PIN23_NotLatched               (0 << 23)
+#define GPIO_LATCH_PIN23_LATCHED                  (1 << 23)
+#define GPIO_LATCH_PIN23_Latched                  (1 << 23)
+#define GPIO_LATCH_PIN24_NOTLATCHED               (0 << 24)
+#define GPIO_LATCH_PIN24_NotLatched               (0 << 24)
+#define GPIO_LATCH_PIN24_LATCHED                  (1 << 24)
+#define GPIO_LATCH_PIN24_Latched                  (1 << 24)
+#define GPIO_LATCH_PIN25_NOTLATCHED               (0 << 25)
+#define GPIO_LATCH_PIN25_NotLatched               (0 << 25)
+#define GPIO_LATCH_PIN25_LATCHED                  (1 << 25)
+#define GPIO_LATCH_PIN25_Latched                  (1 << 25)
+#define GPIO_LATCH_PIN26_NOTLATCHED               (0 << 26)
+#define GPIO_LATCH_PIN26_NotLatched               (0 << 26)
+#define GPIO_LATCH_PIN26_LATCHED                  (1 << 26)
+#define GPIO_LATCH_PIN26_Latched                  (1 << 26)
+#define GPIO_LATCH_PIN27_NOTLATCHED               (0 << 27)
+#define GPIO_LATCH_PIN27_NotLatched               (0 << 27)
+#define GPIO_LATCH_PIN27_LATCHED                  (1 << 27)
+#define GPIO_LATCH_PIN27_Latched                  (1 << 27)
+#define GPIO_LATCH_PIN28_NOTLATCHED               (0 << 28)
+#define GPIO_LATCH_PIN28_NotLatched               (0 << 28)
+#define GPIO_LATCH_PIN28_LATCHED                  (1 << 28)
+#define GPIO_LATCH_PIN28_Latched                  (1 << 28)
+#define GPIO_LATCH_PIN29_NOTLATCHED               (0 << 29)
+#define GPIO_LATCH_PIN29_NotLatched               (0 << 29)
+#define GPIO_LATCH_PIN29_LATCHED                  (1 << 29)
+#define GPIO_LATCH_PIN29_Latched                  (1 << 29)
+#define GPIO_LATCH_PIN30_NOTLATCHED               (0 << 30)
+#define GPIO_LATCH_PIN30_NotLatched               (0 << 30)
+#define GPIO_LATCH_PIN30_LATCHED                  (1 << 30)
+#define GPIO_LATCH_PIN30_Latched                  (1 << 30)
+#define GPIO_LATCH_PIN31_NOTLATCHED               (0 << 31)
+#define GPIO_LATCH_PIN31_NotLatched               (0 << 31)
+#define GPIO_LATCH_PIN31_LATCHED                  (1 << 31)
+#define GPIO_LATCH_PIN31_Latched                  (1 << 31)
+/**@}*/
```

Comment:
so we're at ~800 lines of "on/off << pin number"  Please no.

Reply:
We have defined so that we can stick to database to write a code without seeing header-file. They will be removed for libopencm3-users.